src/main/java/org/fiware/iam/tmforum/NotificationListener.java

```diff
@@ -0,0 +1,92 @@
+package org.fiware.iam.tmforum;
+
+import io.micronaut.http.HttpResponse;
+import io.micronaut.http.annotation.Controller;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.fiware.iam.til.TrustedIssuersListAdapter;
+import org.fiware.iam.tmforum.product.server.api.NotificationListenersClientSideApi;
+import org.fiware.iam.tmforum.product.server.model.*;
+
+import java.util.Collection;
+import java.util.stream.Stream;
+
+@Slf4j
+@Controller("${general.basepath:/}")
+@RequiredArgsConstructor
+public class NotificationListener implements NotificationListenersClientSideApi {
+
+    private final TrustedIssuersListAdapter adapter;
+
+    private final OrganizationResolver organizationResolver;
+
+    @Override
+    public HttpResponse<EventSubscriptionVO> listenToCancelProductOrderCreateEvent(CancelProductOrderCreateEventVO data) {
+        log.warn("Received an unimplemented CancelProductOrderCreateEvent: {}", data);
+        return HttpResponse.ok();
+    }
+
+    @Override
+    public HttpResponse<EventSubscriptionVO> listenToCancelProductOrderInformationRequiredEvent(CancelProductOrderInformationRequiredEventVO data) {
+        log.warn("Received an unimplemented CancelProductOrderInformationRequiredEvent: {}", data);
+        return HttpResponse.ok();
+    }
+
+    @Override
+    public HttpResponse<EventSubscriptionVO> listenToCancelProductOrderStateChangeEvent(CancelProductOrderStateChangeEventVO data) {
+        log.warn("Received an unimplemented CancelProductOrderStateChangeEvent: {}", data);
+        return HttpResponse.ok();
+    }
+
+    @Override
+    public HttpResponse<EventSubscriptionVO> listenToProductOrderAttributeValueChangeEvent(ProductOrderAttributeValueChangeEventVO data) {
+        log.warn("Received an unimplemented ProductOrderAttributeValueChangeEvent: {}", data);
+        return HttpResponse.ok();
+    }
+
+
+    @Override
+    public HttpResponse<EventSubscriptionVO> listenToProductOrderCreateEvent(ProductOrderCreateEventVO data) {
```

Comment:
When the call is dependent on calls to downstream services, its better to use async in micronaut. Since it uses the async-client under the hood, you will see silent timeouts more often than acceptable.